tests/postgres_tests/test_operations.py

```diff
@@ -392,3 +393,64 @@ def test_remove(self):
+        constraint = CheckConstraint(check=Q(pink__gte=4), name='pony_pink_check')
+        new_state = project_state.clone()
+        operation = AddConstraintNotValid(model_name='pony', constraint=constraint)
+        self.assertEqual(
+            operation.describe(),
+            'Create NOT VALID constraint pony_pink_check on model pony'
+        )
+        self.assertEqual(operation.migration_name_fragment, "pony_pony_pink_check_not_valid")
+        operation.state_forwards(self.app_label, new_state)
+        self.assertEqual(len(new_state.models[self.app_label, 'pony'].options['constraints']), 1)
+        self.assertConstraintNotExists(table_name, 'pony_pink_check')
+        # Add constraint.
+        with connection.schema_editor(atomic=True) as editor:
+            operation.database_forwards(self.app_label, editor, project_state, new_state)
+        self.assertConstraintExists(table_name, 'pony_pink_check')
+        # Reversal.
+        with connection.schema_editor(atomic=True) as editor:
+            operation.database_backwards(self.app_label, editor, project_state, new_state)
+        self.assertConstraintNotExists(table_name, ['pony_pink_check'])
+
+    def test_unsupported_constraint(self):
+        constraint = UniqueConstraint(fields=['pink'], name='pony_pink_check')
+        with self.assertRaises(NotSupportedError):
+            AddConstraintNotValid(model_name='pony', constraint=constraint)
+
+
+@unittest.skipUnless(connection.vendor == 'postgresql', 'PostgreSQL specific tests.')
+@modify_settings(INSTALLED_APPS={'append': 'migrations'})
+class ValidateConstraintTests(OperationTestBase):
+    app_label = "test_validate_constraint"
+
+    def test_validate(self):
+        project_state = self.set_up_test_model(self.app_label)
+        table_name = "%s_pony" % self.app_label
+        constraint = CheckConstraint(check=Q(pink__gte=4), name='pony_pink_check')
+        new_state = project_state.clone()
+        operation = AddConstraintNotValid(model_name='pony', constraint=constraint)
+        operation.state_forwards(self.app_label, new_state)
+        with connection.schema_editor(atomic=True) as editor:
+            editor.execute('INSERT INTO %s VALUES(default, -1, 1.2)' % editor.quote_name(table_name))
+            operation.database_forwards(self.app_label, editor, project_state, new_state)
+        new_state = new_state.clone()
+        operation = ValidateConstraint(model_name='pony', name="pony_pink_check")
+        operation.state_forwards(self.app_label, new_state)
+        self.assertEqual(
+            operation.describe(),
+            'Validate constraint pony_pink_check on model pony'
+        )
+        self.assertEqual(operation.migration_name_fragment, "pony_validate_pony_pink_check")
+        with connection.schema_editor(atomic=True) as editor, self.assertRaises(IntegrityError):
+            operation.database_forwards(self.app_label, editor, project_state, new_state)
```

Comment:
Please use `assertRaisesMessage` with the constraint name, to check that the `IntegrityError` actually comes from the constraint - it's easy to trigger `IntegrityError` in many other ways!

Also, can you add a test that succeeds in validating the constraint? Otherwise we aren't checking the success path.